src/crypto/equihash.cpp

```diff
@@ -19,183 +19,210 @@
 #include <iostream>
 #include <stdexcept>
 
-void validate_params(int n, int k)
-{
-    if (k>=n) {
-        std::cerr << "n must be larger than k\n";
-        throw invalid_params();
-    }
-    if (n % 8 != 0) {
-        std::cerr << "Parameters must satisfy n = 0 mod 8\n";
-        throw invalid_params();
-    }
-    if ((n/(k+1)) % 8 != 0) {
-        std::cerr << "Parameters must satisfy n/(k+1) = 0 mod 8\n";
-        throw invalid_params();
-    }
-}
-
-int Equihash::InitialiseState(eh_HashState& base_state)
+template<unsigned int N, unsigned int K>
+int Equihash<N,K>::InitialiseState(eh_HashState& base_state)
 {
+    unsigned int n = N;
+    unsigned int k = K;
     unsigned char personalization[crypto_generichash_blake2b_PERSONALBYTES] = {};
     memcpy(personalization, "ZcashPOW", 8);
     memcpy(personalization+8,  &n, 4);
     memcpy(personalization+12, &k, 4);
```

Comment:
Not related to this PR: Here's another place we're native endian dependent (#850).